### PR TITLE
ARTEMIS-1577 Address-settings policies not working with older clients

### PR DESCRIPTION
**Pull request by clebertsuconic**
This closes #1744

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java [hunk 1]
```diff
@@ -736,11 +736,11 @@ public RoutingStatus route(final Message message,
          throw new IllegalStateException("Message cannot be routed more than once");
       }
 
-      setPagingStore(message);
+      setPagingStore(context.getAddress(message), message);
 
       AtomicBoolean startedTX = new AtomicBoolean(false);
 
-      final SimpleString address = message.getAddressSimpleString();
+      final SimpleString address = context.getAddress(message);
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: Can we not make the call to context.getAddress just the once, and re-use the result throughout the method call?

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java [hunk 2]
```diff
@@ -1284,7 +1284,7 @@ private boolean checkDuplicateID(final Message message,
          boolean isDuplicate = false;
 
          if (duplicateIDBytes != null) {
-            cache = getDuplicateIDCache(message.getAddressSimpleString());
+            cache = getDuplicateIDCache(context.getAddress(message));
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: Same as other comment, re making one call to context.getAddress(message) and re-using the result

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java (hunk 1 above)
Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: I'm just going for the simple refactoring here.. every call to message.getAddressSimplString() to be calling context.getAddress()

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: Your comment could apply to the previous version as well.. I didn't make any big changes here. nor did want to risk anything else.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java (hunk 2 above)
Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: @michaelandrepearce caching here goes beyond the scope here.

I would need to change a lot of signatures to pass an address as a parameter.. or cache within each method used during routing.


We can do that.. but it goes beyond the scope of this fix.

Comment (michaelandrepearce) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: No worries I can do separate pr later

**Comment by michaelandrepearce** [comment]
+1 I’m happy as noted I can pr some bits later

**Pull request closed without merging**